### PR TITLE
tests: Fix alignment

### LINKED ISSUE #44259: intel_adsp_cavs18: tests/lib/icmsg failed
**Describe the bug**
tests/lib/icmsg failed on intel_adsp_cavs18

**To Reproduce**
1. twister -W -p intel_adsp_cavs18 --device-testing --device-serial-pty=$HOME/zephyrproject/zephyr/soc/xtensa/intel_adsp/tools/cavstwist.sh --west-flash=$HOME/zephyrproject/zephyr/soc/xtensa/intel_adsp/tools/cavstwist.sh -T tests/lib/icmsg/ -vvv

**Logs and console output**
```
INFO:cavs-fw:Mapped PCI bar 0 of length 16384 bytes.
INFO:cavs-fw:Selected output stream 7 (GCAP = 0x1009701)
INFO:cavs-fw:Mapped PCI bar 4 of length 1048576 bytes.
INFO:cavs-fw:Detected cAVS 1.8+ hardware
INFO:cavs-fw:Trimming 16 bytes of extended manifest
INFO:cavs-fw:Resetting HDA device
INFO:cavs-fw:Powering down DSP cores
INFO:cavs-fw:Configuring HDA stream 7 to transfer firmware image
INFO:cavs-fw:Mapped 2M huge page at 0x140a00000 to contain 81920 bytes of firmware
INFO:cavs-fw:Filled the buffer descriptor list (BDL) for DMA.
INFO:cavs-fw:Starting DSP, ADSPCS = 0xf0f
INFO:cavs-fw:Wait for ROM startup
INFO:cavs-fw:Sending IPC command, HIPIDR = 0x81004000
INFO:cavs-fw:Starting DMA, FW_STATUS = 0x5000001
INFO:cavs-fw:Waiting for firmware handoff, FW_STATUS = 0x5000001
INFO:cavs-fw:cAVS firmware load complete
--
Running test suite icmsg_buf
===================================================================
START - test_icmsg_buf_ut
E:  ** FATAL EXCEPTION
E:  ** CPU 0 EXCCAUSE 9 (load/store alignment)
E:  **  PC 0xbe0117b7 VADDR 0x9e01b3c2
E:  **  PS 0x60020
E:  **    (INTLEVEL:0 EXCM: 0 UM:1 RING:0 WOE:1 OWB:0 CALLINC:2)
E:  **  A0 0xbe010914  SP 0xbe018ec0  A2 0x9e01b3c2  A3 0xc8
E:  **  A4 0xe  A5 (nil)  A6 0x9e01b000  A7 0xbe017517
E:  **  A8 0xffffffff  A9 0xbe0123d4 A10 0x9e018300 A11 0xffffffff
E:  ** A12 0xbe019e60 A13 0xbe019e40 A14 0x9e01b000 A15 0x9e01b2dc
E:  ** LBEG (nil) LEND (nil) LCOUNT (nil)
E:  ** SAR (nil)
E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
E: Current thread: 0x9e01b000 (test_icmsg_buf_ut)
E: Halting system
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain zephyr-sdk-0.13.2
 - Commit 6a5520c62693

### PR DESCRIPTION
**Pull request by eobalski**
@LixinGuoX could you verify if this fixes #44259?

### CODE REVIEW

**Comment by emob-nordic** [comment]
@aspeeddylan I saw you added
[default CACHE_TYPE as HAS_EXTERNAL_CACHE](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/aspeed/ast10x0/Kconfig.defconfig.series#L18:L20). In this PR the test that uses cache management is run and it fails due to undefined reference to `cache_data_range`.
```
[...]/ast1030_evb/tests/subsys/ipc/icmsg_buf/subsys.ipc.icmsg_buf/../../../../../../../include/cache.h:91: undefined reference to `cache_data_range'
```
Could you please verify if `HAS_EXTERNAL_CACHE` should be selected for this particular soc?
It doesn't work with `HAS_ARCH_CACHE` either.
Could you please have a look? Should i exclude this platform from this test?

**Comment by nashif** [comment]
@emob-nordic this fixes the issue, but please fix the commit message: "s/a" is as bad as not having anything in the commit body.

**Comment by aspeeddylan** [comment]
Hi Eobalski,

This SOC has an external cache, but the cache driver is not included in this PR.  I will create another PR for the external cache driver in drivers/cache/, and the missing reference ‘cache_data_range’ will also be added.  Before this, could you exclude the platform for the cache test?  Thank you very much.

--
Dylan

寄件者: eobalski ***@***.***>
日期: 星期四, 2022年3月31日 下午6:59
收件者: zephyrproject-rtos/zephyr ***@***.***>
副本: Dylan Hung ***@***.***>, Mention ***@***.***>
主旨: Re: [zephyrproject-rtos/zephyr] tests: Fix alignment (PR #44401)

@aspeeddylan<https://github.com/aspeeddylan> I saw you added
default CACHE_TYPE as HAS_EXTERNAL_CACHE<https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/aspeed/ast10x0/Kconfig.defconfig.series#L18:L20>. In this PR the test that uses cache management is run and it fails due to undefined reference to cache_data_range.

[...]/ast1030_evb/tests/subsys/ipc/icmsg_buf/subsys.ipc.icmsg_buf/../../../../../../../include/cache.h:91: undefined reference to `cache_data_range'

Could you please verify if HAS_EXTERNAL_CACHE should be selected for this particular soc?
It doesn't work with HAS_ARCH_CACHE either.
Could you please have a look? Should i exclude this platform from this test?

**Comment by LixinGuoX** [comment]
Verified on intel_adsp_cavs18.

**Comment by emob-nordic** [comment]
@LixinGuoX @nashif  could you have a look, accept and merge?

**Review by ghost — DISMISSED** [review]

File: tests/lib/icmsg/src/main.c [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * 212 - sizeof(struct icmsg_buf) - 1 = 199.
  * -1 because internal rd/wr_idx is reserved to mean the buffer is empty.
  */
-static uint8_t memory_area[212];
+static uint8_t memory_area[212] __aligned(4);
```
Comment (mbolivar-nordic) [inline tests/lib/icmsg/src/main.c:14]: The commit log is just describing the changes without providing any justification for them. Why?

**Review by mbolivar-nordic — DISMISSED** [review]

**Comment by emob-nordic** [comment]
@mbolivar-nordic i updated the commit message, is that okay now?

File: tests/lib/icmsg/src/main.c (hunk 1 above)
Comment (mbolivar-nordic) [reply tests/lib/icmsg/src/main.c:14]: Thanks for the updated commit log. I'm going to drop my NAK.

**Review by nashif — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Pull request merged**